tests/unittests/text/test_perplexity.py

```diff
@@ -80,3 +80,20 @@ def test_accuracy_differentiability(self, preds, target, ignore_index):
             metric_functional=perplexity,
             metric_args={"ignore_index": ignore_index},
         )
+
+    @pytest.mark.parametrize("dtype", [torch.half, torch.double])
+    def test_perplexity_dtypes_cpu(self, preds, target, ignore_index, dtype):
+        """Test dtype support of the metric on CPU."""
+        if dtype == torch.half:
+            pytest.skip("`softmax_lastdim_kernel_impl` is not support for half precision on CPU.")
```

Comment:
I think this should be a `pytest.raises()` to assert we raise the appropriate error. Otherwise, if you skip it then you could also just remove it from the parameterization.